src/handlers/review_prefs.rs

```diff
@@ -0,0 +1,304 @@
+    let rec = db
+        .query_one(
+            q,
+            &[
+                &prefs.user_id,
+                &prefs.max_assigned_prs,
+                &prefs.pto_date_start,
+                &prefs.pto_date_end,
+                &prefs.active,
+                &prefs.allow_ping_after_days,
+                &prefs.publish_prefs,
+            ],
+        )
+        .await
+        .context("Update DB error")?;
+    Ok(rec.into())
+}
+
+/// Return a team member identified by a checksum
+pub async fn get_user(db_client: &DbClient, checksum: &str) -> anyhow::Result<ReviewCapacityUser> {
+    let q = "
+SELECT username,r.*
+FROM review_capacity r
+JOIN users on r.user_id=users.user_id
+WHERE r.checksum=$1";
+    let rec = db_client
+        .query_one(q, &[&checksum])
+        .await
+        .context("SQL error")?;
+    Ok(rec.into())
+}
+
+/// Get review preferences for a number of team members
+/// - me: sort the current user at the top of the list
+/// - is_admin: if `true` return also preferences marked as not public
+pub async fn get_prefs(
+    db: &DbClient,
+    users: &mut Vec<String>,
+    me: &str,
+    is_admin: bool,
+) -> Vec<ReviewCapacityUser> {
+    let q = format!(
+        "
+SELECT username,r.*
+FROM review_capacity r
+JOIN users on r.user_id=users.user_id
+WHERE username = any($1)
+ORDER BY case when username='{}' then 1 else 2 end, username;",
+        me
+    );
+
+    let rows = db.query(&q, &[users]).await.unwrap();
+    rows.into_iter()
+        .filter_map(|row| {
+            let rec = ReviewCapacityUser::from(row);
+            if rec.username == me || rec.publish_prefs || (!rec.publish_prefs && is_admin) {
+                Some(rec)
+            } else {
+                None
+            }
+        })
+        .collect()
+}
+
+/// Get review preferences for a number of team members
+pub async fn get_review_candidates_by_username(
+    db: &DbClient,
+    usernames: Vec<String>,
+) -> anyhow::Result<Vec<ReviewCapacityUser>> {
+    let q = format!(
+        "
+SELECT u.username, r.*
+FROM review_capacity r
+JOIN users u on u.user_id=r.user_id
+WHERE username = ANY('{{ {} }}')",
+        usernames.join(",")
+    );
+    let rows = db.query(&q, &[]).await.unwrap();
+    Ok(rows
+        .into_iter()
+        .filter_map(|row| Some(ReviewCapacityUser::from(row)))
+        .collect())
+}
+
+/// Get review preferences for a number of team members, filter by review capacity
+pub async fn get_review_candidate_by_capacity(
+    db: &DbClient,
+    usernames: Vec<String>,
+) -> anyhow::Result<ReviewCapacityUser> {
+    let q = format!(
+        "
+SELECT username, r.*, sum(r.max_assigned_prs - r.num_assigned_prs) as avail_slots
+FROM review_capacity r
+JOIN users on users.user_id=r.user_id
+WHERE active = true
+AND current_date NOT BETWEEN pto_date_start AND pto_date_end
+AND username = ANY('{{ {} }}')
+AND num_assigned_prs < max_assigned_prs
+GROUP BY username, r.id
+ORDER BY avail_slots DESC
+LIMIT 1",
```

Comment:
The core of the new PR assignment is here: This SQL query will retrieve all active team members and sort them by "avail slots" (`max # of PRs` - `# of PRs assigned`)